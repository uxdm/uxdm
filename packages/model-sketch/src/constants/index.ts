export * from './layout';
export * from './styles';
