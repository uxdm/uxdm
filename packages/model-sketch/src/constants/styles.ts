export const SYSTEM_FONTS = [
  /* * Apple * */
  '-apple-system',
  'system-ui',
  'BlinkMacSystemFont',
  /* * Microsoft * */
  'Segoe UI',
  /* * Android * */
  'Roboto',
];

type FontWeightType =
  | 'Regular'
  | 'Bold'
  | 'Semibold'
  | 'UltraLight'
  | 'Thin'
  | 'Light'
  | 'Medium'
  | 'Heavy'
  | 'Black';

type FontWeightEnum = {
  normal: FontWeightType;
  bold: FontWeightType;
  bolder: FontWeightType;
  '100': FontWeightType;
  '200': FontWeightType;
  '300': FontWeightType;
  '400': FontWeightType;
  '500': FontWeightType;
  '600': FontWeightType;
  '700': FontWeightType;
  '800': FontWeightType;
  '900': FontWeightType;
};

/**
 * 字体权重
 * */
export const FONT_WEIGHTS: FontWeightEnum = {
  normal: 'Regular',
  bold: 'Bold',
  bolder: 'Semibold',
  '100': 'UltraLight',
  '200': 'Thin',
  '300': 'Light',
  '400': 'Regular',
  '500': 'Medium',
  '600': 'Semibold',
  '700': 'Bold',
  '800': 'Heavy',
  '900': 'Black',
};
