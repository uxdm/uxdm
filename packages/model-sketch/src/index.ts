export { Frame } from './objects';
export { Group, Rectangle } from './layers';

export * from './types';
export * from './constants';
