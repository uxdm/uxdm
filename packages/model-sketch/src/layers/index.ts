export * from './Rectangle';
export * from './Group';
