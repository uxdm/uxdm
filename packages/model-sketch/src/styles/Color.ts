import { Color as BaseColor } from 'uxdm';
import { SketchFormat } from '../types';

/**
 * 针对 Sketch 的颜色类
 */
class Color extends BaseColor {
  /**
   * 转为 Sketch JSON对象
   * @returns {SketchFormat.Color} color json
   */
  toSketchJSON = (): SketchFormat.Color => {
    return {
      _class: 'color',
      red: this.red / 255,
      green: this.green / 255,
      blue: this.blue / 255,
      alpha: this.alpha,
    };
  };
}

export default Color;
