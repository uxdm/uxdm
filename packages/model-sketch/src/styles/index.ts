export { default as Color } from './Color';
export { default as Shadow } from './Shadow';
export { default as Border } from './Border';
export { default as Gradient } from './Gradient';
export { default as Fill } from './Fill';
export { default as Image } from './Image';
export { default as Style } from './Style';
export { default as TextStyle } from './TextStyle';
