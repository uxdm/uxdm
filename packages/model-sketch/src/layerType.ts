import { AbstractNodeParams, RectangleNodeParams } from '@uxdm/schema';
import { Group } from './layers/Group';
import { Rectangle } from './layers/Rectangle';
import { Assign } from 'utility-types';

/**
 * 任意可以成为图层的对象
 */
export type AnyLayer = Group | Rectangle;

/**
 * 抽象节点的入参
 */
export type SketchLayerParams = AbstractNodeParams;

export type SketchGroupParams = Assign<
  SketchLayerParams,
  { children?: AnyLayer[] }
>;

export type SketchRectangleParams = Assign<
  AbstractNodeParams,
  RectangleNodeParams
>;
