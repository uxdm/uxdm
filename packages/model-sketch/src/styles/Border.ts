import { BorderParams } from '@uxdm/schema';
import { Border as BaseBorder } from 'uxdm';
import Color from './Color';
import Gradient from './Gradient';
import { SketchFormat } from '../types';
import { getContextSettings, getFillType } from '../utils';

/**
 * 描边对象
 * */
class Border extends BaseBorder {
  constructor(params?: BorderParams) {
    super(params);
    if (params) {
      const { color, gradient } = params;

      this.color = new Color(color);
      this.gradient = new Gradient(gradient);
    }
  }

  color: Color = new Color();

  gradient: Gradient = new Gradient();

  get sketchPosition() {
    switch (this.align) {
      default:
      case 'INSIDE':
        return SketchFormat.BorderPosition.Inside;
      case 'CENTER':
        return SketchFormat.BorderPosition.Center;
      case 'OUTSIDE':
        return SketchFormat.BorderPosition.Outside;
    }
  }

  /**
   * 转为 Sketch JSON 对象
   * @returns {SketchFormat.Border}
   */
  toSketchJSON = (): SketchFormat.Border => {
    return {
      _class: SketchFormat.ClassValue.Border,
      isEnabled: this.visible,
      fillType: getFillType(this.type),
      color: this.color.toSketchJSON(),
      contextSettings: getContextSettings(this.blendMode, this.opacity),
      gradient: this.gradient.toSketchJSON(),
      position: this.sketchPosition,
      thickness: this.thickness,
    };
  };
}

export default Border;
