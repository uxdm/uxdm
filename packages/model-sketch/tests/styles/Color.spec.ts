import { Color } from '../../src/styles';

describe('Color 类', () => {
  describe('toSketchJSON', () => {
    it('正常解析', () => {
      const color = new Color();

      expect(color.toSketchJSON()).toMatchSnapshot();
    });
  });
});
