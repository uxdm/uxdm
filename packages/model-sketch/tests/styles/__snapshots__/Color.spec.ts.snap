// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color 类 toSketchJSON 正常解析 1`] = `
Object {
  "_class": "color",
  "alpha": 1,
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;
